### ISSUE TITLE
DownloadUrlResolver won't return any .flv, unable to extract audio as a result.

### ISSUE THREAD

**Issue by trineroks** [flagbug/YoutubeExtractor#191]
I came across a very very strange issue. YoutubeExtractor was working perfectly fine up until today - for whatever reason the DownloadUrlResolver won't or can't get any .flv videos so I am unable to extract audio.

For example, this video https://www.youtube.com/watch?v=LODkVkpaVQA was working perfectly fine yesterday. Now when I try to do the same today I get this:

http://i.imgur.com/D1YMwd3.png

There are no .flv videos in this list. The last 3 videos are, for some reason, totally empty. I've tested this url and multiple others that have worked yesterday through the ExampleApplication provided by YoutubeExtractor, but I get the same error there as well.

**Comment by flagbug** [comment]
YouTube is phasing out Flash videos, so the format might not be supported anymore

**Comment by trineroks** [comment]
Any other way to extract audio then?

**Comment by flagbug** [comment]
Yes, use [FFMPEG](https://ffmpeg.org/) with any of the other formats

**Comment by Kimmax** [comment]
Also one should consider looking into the others formats anyway, since we have some audio only streams, which don't need any extraction at all, making thinks faster and cleaner

**Comment by trineroks** [comment]
@Kimmax could you elaborate on that? Is there a way I could extract strictly mp3/AAC audio from a Youtube video without having to download the video?

How I've been doing it with YoutubeExtractor was I modified the source code such that I would be able to download the video into a Stream -> extract mp3 -> return the mp3 as a MemoryStream which I would then play back. Are you saying it's possible to completely skip the video download/extract audio step and just download the audio track for any Youtube video?

**Comment by Kimmax** [comment]
Already done [here](https://github.com/Kimmax/SYMMExtractor/blob/master/YoutubeExtractor/YoutubeExtractor/AudioStreamer.cs).
You just have to select the right file using [some Linq magic ](https://github.com/Kimmax/SYMM/blob/master/src/Downloader.cs#L143) and pass the resulting URL to some player which is able to stream http resources.  You can implement one yourself or use one of the various librarys that already can, e.g. I use [NAudio](https://github.com/naudio/NAudio) for this which works flawlessly.
For the download part:
I just throw everything into ffmpeg and set the desired output format. Ffmpeg detects if the input is the same as output and just copies it over, so there is almost no loss of speed.

**Comment by madfisht3** [comment]
It's very bad! I exclude extract audio function from my software :(

**Comment by Kimmax** [comment]
@madfisht3 I don't know what you mean. @flagbug mentioned earlier that he basically doesn't even want the extraction code in this project and you have a perfectly performant working example in my fork which converts the video in almost every format you can imagine

**Comment by trineroks** [comment]
VideoInfo videoInfo = videoInfos
                        .Where(info => info.CanExtractAudio && info.AudioBitrate > 0)
                        .OrderBy(info => info.AudioBitrate)
                        .OrderBy(info => info.AdaptiveType == AdaptiveType.Audio)
                        .OrderBy(info => info.AudioBitrate == settings.AudioBitrate)
                        .Last();

This code in Downloader.cs won't work, since my entire issue is that DownloadUrlResolver can't find any .flv videos, and info.CanExtractAudio pretty much only returns if a video is .flv or not.

**Comment by Kimmax** [comment]
Jup you're right,  I patched the `CanExtraxtAudio` prop on my fork. Forgot to mention that

**Comment by Kimmax** [comment]
Also note that you have to replace the AudioExtractor itself since the current one is only able to extract from flv

**Comment by candidodmv** [comment]
Hello @flagbug ,
So, even YouTube is phasing out Flash videos, this library was adapted to handling other video types for keeping extract audio support?

**Comment by Kimmax** [comment]
@candidodmv see above:
> @flagbug mentioned earlier that he basically doesn't even want the extraction code in this project

**Comment by candidodmv** [comment]
@Kimmax  Why not? In short time ago(until YouTube is phasing out Flash videos) this project did  do this ? Now, is necessary to adapt for handling another available formats(if doesn't someone already did). This is the most obivious, of course, that I think.

**Comment by slavbar** [comment]
@Kimmax I think that you meant

VideoInfo videoInfo = videoInfos
.Where(info => info.CanExtractAudio && info.AudioBitrate > 0)
.OrderBy(info => info.AudioBitrate)
.**ThenBy**(info => info.AdaptiveType == AdaptiveType.Audio)
.**ThenBy**(info => info.AudioBitrate == settings.AudioBitrate)
.Last();

**Comment by Kimmax** [comment]
That's quite a while ago, I don't really remember what I did there 🗡 
Nevertheless I think it should be `.AndThen()` not `.ThenBy()`
